### PR TITLE
Chapter 6: Your Second Activity

### PR DESCRIPTION
**Pull request by OnicaAdrianCatalin**
## New features

 Pressing cheat button will bring you to another activity in which the correct answer will be displayed.

 If you try to answer again the question you'll be punished with a judgment toast.

## Changes in code

Added a new activity.

Used intent extras to pass data between activities.

https://user-images.githubusercontent.com/102966412/163408379-01653a1e-4af6-459e-984b-ff9b7d35c961.mov

### CODE REVIEW

File: app/src/main/java/com/example/geoquiz/CheatActivity.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+
+const val EXTRA_ANSWER_IS_TRUE = "com.example.geoquiz.answer_is_true"
+const val EXTRA_ANSWER_SHOWN = "com.example.geoquiz.answer_shown"
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/CheatActivity.kt]: These 2 const vals can be moved to the companion object

File: app/src/main/java/com/example/geoquiz/QuizViewModel.kt [hunk 2]
```diff
@@ -4,6 +4,7 @@ import androidx.lifecycle.ViewModel
 
 class QuizViewModel : ViewModel() {
 
+
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/QuizViewModel.kt]: One empty line here is more than enough

File: app/src/main/java/com/example/geoquiz/MainActivity.kt [hunk 3]
```diff
@@ -47,6 +51,12 @@ class MainActivity : AppCompatActivity() {
             Toast.makeText(this, quizViewModel.checkAnswer(false), Toast.LENGTH_SHORT).show()
         }
 
+        cheatButton.setOnClickListener {
+            val answerIsTrue = quizViewModel.currentQuestionAnswer
```
Comment (iugaliviu) [inline app/src/main/java/com/example/geoquiz/MainActivity.kt]: I would rename this val to `answerQuestion` or `currentAnswerQuestion` to keep consistency with `quizViewModel.currentQuestionAnswer`

File: app/src/main/java/com/example/geoquiz/CheatActivity.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+class CheatActivity : AppCompatActivity() {
+    private lateinit var answerTextView: TextView
+    private lateinit var showAnswerButton: Button
+    private var answerQuestion = false
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_cheat)
+
+        answerTextView = findViewById(R.id.answer_text_view)
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/CheatActivity.kt]: As a personal preference, I would move the `findViewById`s to a method called `bindViews`

File: app/src/main/java/com/example/geoquiz/CheatActivity.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        showAnswerButton = findViewById(R.id.show_answer_button)
+        answerQuestion = intent.getBooleanExtra(EXTRA_ANSWER_IS_TRUE, false)
+
+        setOnClickListeners()
+    }
+
+    companion object {
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/CheatActivity.kt]: companion objects are usually defined at the bottom https://discuss.kotlinlang.org/t/companion-objects-on-bottom/17797

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -6,6 +6,10 @@
     <string name="correct">Correct!</string>
     <string name="incorrect">Incorrect!</string>
     <string name="next">Next</string>
+    <string name="warning">Are you sure you want to do this?</string>
```
Comment (NegreaVlad) [inline app/src/main/res/values/strings.xml:9]: good names

File: app/src/main/java/com/example/geoquiz/MainActivity.kt [hunk 7]
```diff
@@ -47,6 +51,12 @@ class MainActivity : AppCompatActivity() {
             Toast.makeText(this, quizViewModel.checkAnswer(false), Toast.LENGTH_SHORT).show()
         }
 
+        cheatButton.setOnClickListener {
+            val answerQuestion = quizViewModel.currentQuestionAnswer
+            val intent = CheatActivity.newIntent(this@MainActivity, answerQuestion)
+            startActivityForResult(intent, REQUEST_CODE_CHEAT)
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/MainActivity.kt]: @iugaliviu mentioned that `startActivityForResult` is deprecated
Please use the new method:  https://stackoverflow.com/questions/62671106/onactivityresult-method-is-deprecated-what-is-the-alternative

**Review by NegreaVlad — COMMENTED** [review]
Very good overall

File: app/src/main/java/com/example/geoquiz/CheatActivity.kt [hunk 8]
```diff
@@ -52,4 +39,20 @@ class CheatActivity : AppCompatActivity() {
         }
         setResult(Activity.RESULT_OK, data)
     }
+
+    companion object {
```
Comment (NegreaVlad) [inline app/src/main/java/com/example/geoquiz/CheatActivity.kt:48]: Please always place companion objects at the bottom.

**Review by NegreaVlad — APPROVED** [review]
LGTM! Small change required.

**Pull request merged**